sycl/include/CL/sycl/handler.hpp

```diff
@@ -551,6 +551,151 @@ class __SYCL_EXPORT handler {
+        KernelFunc);
+  }
+
+  // For 'sycl::nd_item<Dims>' kernel argument
+  template <class KernelType, typename ArgT, int Dims>
+  typename std::enable_if<std::is_same<ArgT, sycl::nd_item<Dims>>::value,
+                          KernelType *>::type
+  ResetHostKernel(const KernelType &KernelFunc) {
+    struct NormalizedKernelType {
+      KernelType MKernelFunc;
+      NormalizedKernelType(const KernelType &KernelFunc)
+          : MKernelFunc(KernelFunc) {}
+      void operator()(const nd_item<Dims> &Arg) {
+        detail::runKernelWithArg(MKernelFunc, Arg);
+      }
+    };
+    return ResetHostKernelHelper<KernelType, struct NormalizedKernelType, Dims>(
+        KernelFunc);
+  }
+
+  // For 'sycl::item<Dims, without_offset>' kernel argument
+  template <class KernelType, typename ArgT, int Dims>
+  typename std::enable_if<std::is_same<ArgT, sycl::item<Dims, false>>::value,
+                          KernelType *>::type
+  ResetHostKernel(const KernelType &KernelFunc) {
+    struct NormalizedKernelType {
+      KernelType MKernelFunc;
+      NormalizedKernelType(const KernelType &KernelFunc)
+          : MKernelFunc(KernelFunc) {}
+      void operator()(const nd_item<Dims> &Arg) {
+        sycl::item<Dims, false> Item = detail::Builder::createItem<Dims, false>(
+            Arg.get_global_range(), Arg.get_global_id());
+        detail::runKernelWithArg(MKernelFunc, Item);
+      }
+    };
+    return ResetHostKernelHelper<KernelType, struct NormalizedKernelType, Dims>(
+        KernelFunc);
+  }
+
+  // For 'sycl::item<Dims, with_offset>' kernel argument
+  template <class KernelType, typename ArgT, int Dims>
+  typename std::enable_if<std::is_same<ArgT, sycl::item<Dims, true>>::value,
+                          KernelType *>::type
+  ResetHostKernel(const KernelType &KernelFunc) {
+    struct NormalizedKernelType {
+      KernelType MKernelFunc;
+      NormalizedKernelType(const KernelType &KernelFunc)
+          : MKernelFunc(KernelFunc) {}
+      void operator()(const nd_item<Dims> &Arg) {
+        sycl::item<Dims, true> Item = detail::Builder::createItem<Dims, true>(
+            Arg.get_global_range(), Arg.get_global_id(), Arg.get_offset());
+        detail::runKernelWithArg(MKernelFunc, Item);
+      }
+    };
+    return ResetHostKernelHelper<KernelType, struct NormalizedKernelType, Dims>(
+        KernelFunc);
+  }
+
+  // For 'sycl::group<Dims>' kernel argument
+  template <class KernelType, typename ArgT, int Dims>
+  typename std::enable_if<std::is_same<ArgT, sycl::group<Dims>>::value,
+                          KernelType *>::type
+  ResetHostKernel(const KernelType &KernelFunc) {
+    struct NormalizedKernelType {
+      KernelType MKernelFunc;
+      NormalizedKernelType(const KernelType &KernelFunc)
+          : MKernelFunc(KernelFunc) {}
+      void operator()(const nd_item<Dims> &Arg) {
+        detail::runKernelWithArg(MKernelFunc, Arg.get_group());
```

Comment:
This version is supposed to be used with parallel_for_workgroup, which is not supported for ESIMD. So I'd make operator() just throw, and comment out `detail::runKernelWithArg(MKernelFunc, Arg.get_group());`

Reply:
Following tests could fail without this change even if esimd_cpu build is disable by default. This is because this file is used for all backends and host kernel. I applied this change because of them.

https://github.com/intel/llvm-test-suite/blob/intel/SYCL/Basic/stream/stream.cpp
https://github.com/intel/llvm-test-suite/blob/intel/SYCL/SpecConstants/2020/kernel_lambda_with_kernel_handler_arg.cpp

Reply:
Tests containing parallel_for_work_group should not be run with ESIMD, as they will fail. You may exclude those tests for ESIMD_CPU_EMU. @vladimirlaz, can we create a lit feature for ESIMD CPU EMU plugin similar to opencl and level zero?

Reply:
I'll revert changes related to this 'group' argument support after a lit feature for ESIMD CPU Emulation plug-in is created.

Reply:
> I'll revert changes related to this 'group' argument support after a lit feature for ESIMD CPU Emulation plug-in is created.

Am I right here? Should `ResetHostKernel` also have to cover `sycl::group<>` argument type because `MHostKernel.reset()` from left side is changed with call to `ResetHostKernel`?

Reply:
@vladimirlaz / @kbobrovs , are we going to create a lit feature for ESIMD_CPU emulation for this type of exclusion?

And, how did tests with `parallel_for_work_groups` pass for ESIMD so far?

Reply:
I would like to highlight that on-device tests (the tests which actually executed on target BE) have been removed from the intel/llvm repo to intel/llvm-test-suite (see https://github.com/intel/llvm/pull/4393).
So, these test should be migrated to intel/llvm-test-suite. Correspondign feature can be added the (see HIP BE as an example).

Reply:
@vladimirlaz, the tests I mentioned above are already in intel/llvm-test-suite. You can see in their URLs.

Reply:
@dongkyunahn-intel you can create LIT feature esimd_cpu and use it for the tests (like @kbobrovs suggested).

Reply:
@vladimirlaz , what file should I update for creating a LIT feature?